### ISSUE TITLE
Website has some issues on mobile

### ISSUE THREAD

**Issue by benmajor** [gabordemooij/redbean#890]
When visiting the website on a mobile device (specifically iPhone 13 Pro Max using Chrome), there seems to be an issue with the menu overlay. 

The overlay seems to be shown by default, and attempts to dismiss it (i.e. tapping in the overlay) just reveal the menu. Screenshots as follows:

![image](https://user-images.githubusercontent.com/1329832/158011778-19b80455-f7f6-47a7-ba91-a5482d8f7073.png)

![image](https://user-images.githubusercontent.com/1329832/158011801-3a6f9bdf-bd32-4783-aef8-e86edd0cba92.png)

**Comment by gabordemooij** [comment]
Hi, thanks for reporting.
Unfortunately I can only emulate Safari using Midori (since I work on Linux/BSD only) and Midori works fine.
So I guess this must be something new. Have to look into this....

**Comment by benmajor** [comment]
No problem, I've had a quick look, but couldn't see the exact cause. Interestingly, using Chrome Dev Tools on desktop seems to overflow the main container, so I suspect the issue lies somewhere there.

**Comment by gabordemooij** [comment]
Is it better now?

**Comment by benmajor** [comment]
No, sadly not. It's still exhibiting the same behaviour.

**Comment by sah-comp** [comment]
if has something to do with code in mobile.js

```

function iOS() {

  var iDevices = [
    'iPad Simulator',
    'iPhone Simulator',
    'iPod Simulator',
    'iPad',
    'iPhone',
    'iPod'
  ];

  if (!!navigator.platform) {
    while (iDevices.length) {
      if (navigator.platform === iDevices.pop()){ return true; }
    }
  }

  return false;
}

if (iOS()) {
	document.querySelector('.wrapper').style  = '-webkit-transform: translate3d(0,0,0);';
}
```

When the .wrapper is styled like the observed appearance takes place.

**Comment by gabordemooij** [comment]
Ok, that might be helpful, thanks

**Comment by gabordemooij** [comment]
Sorry for this. Ok, I changed the script to also check for Safari, this was obviously a fix for iOS/Safari.
Unfortunately I don't have any Apple/iOS products myself so it's hard to test for me. This is why I use these ugly fixes, if I had a test device I would have fixed the CSS itself, but now I have to rely on 3rd party fixes.

**Comment by gabordemooij** [comment]
Does it work now, or does it still have the issue? I now check for the word Safari in the userAgent string. Awful, but the only way, without any test device.

**Comment by benmajor** [comment]
Are you able to post the website into a repo? I'm happy to take a look at a possible fix.

**Comment by benmajor** [comment]
> Does it work now, or does it still have the issue? I now check for the word Safari in the userAgent string. Awful, but the only way, without any test device.

I don't think that the check for Safari is the issue here, as it persist in Chrome, too.

**Comment by sah-comp** [comment]
> > Does it work now, or does it still have the issue? I now check for the word Safari in the userAgent string. Awful, but the only way, without any test device.
> 
> I don't think that the check for Safari is the issue here, as the issue persists in Chrome, too. Sadly, it doesn't appear to have resolved the issue at least.

Not the check for Safari, but setting the style of .wrapper to -webkit-transform: translate3d(0,0,0)

**Comment by gabordemooij** [comment]
> > > Does it work now, or does it still have the issue? I now check for the word Safari in the userAgent string. Awful, but the only way, without any test device.
> > 
> > 
> > I don't think that the check for Safari is the issue here, as the issue persists in Chrome, too. Sadly, it doesn't appear to have resolved the issue at least.
> 
> Not the check for Safari, but setting the style of .wrapper to -webkit-transform: translate3d(0,0,0)

Yes, but that style will only be applied if Safari is detected in the new JS file. Maybe hard refresh?

**Comment by gabordemooij** [comment]
I tested the site with a user agent switcher and set the user agent to Chome on iOS. The translate3d-fix for iOS/Safari that causes the issue is no longer applied over here.

**Comment by gabordemooij** [comment]
Okay, for now I just disabled the Safari-fix, that seems to break things in both Safari and Chrome. I hope this improves the user experience somewhat. I tested it in app.lambdatest.com and it seems to work.There is no menu button in the upper corner though, but there is a menu page as a fallback. Not sure what causes these issues on iOS devices. However, testing is quite hard. Even with lambdatest. I can barely scroll in the iPhone and the test system is very slow.

**Comment by sah-comp** [comment]
One can tap on upper right corner to get the menu, although it would not win a prize as intuitive user experience ;-)

I +1 benmajor for asking to put the site in a repo. I am also willing to have a look at it.

**Comment by gabordemooij** [comment]
Okay, the issue seems to stem from the fact that the menu button has been defined in an ::after element, I rewired this. The experience must be better now.

**Comment by gabordemooij** [comment]
![image](https://user-images.githubusercontent.com/89826/158562586-4b7f513f-d9a9-4604-be06-5941a01b46dd.png)
![image](https://user-images.githubusercontent.com/89826/158562643-09bbd8d5-dbc8-4bce-90bb-becd2da9500b.png)
![image](https://user-images.githubusercontent.com/89826/158562934-51e90fcd-5df9-41b5-9046-32fd7f3e9d9d.png)
![image](https://user-images.githubusercontent.com/89826/158562984-1059bb1f-55f6-4a12-be46-08544f2edf7b.png)

**Comment by sah-comp** [comment]
I can confirm this works well on iPhone Safari now.

**Comment by gabordemooij** [comment]
Okay, great. Thank you for your patience @sah-comp @benmajor

**Comment by benmajor** [comment]
Yes, I can confirm it's also fixed in Chrome:

![image](https://user-images.githubusercontent.com/1329832/158570102-02ef7772-4591-4a02-af30-d9aaaa96f266.png)

Thanks for your efforts, @gabordemooij!

**Comment by sah-comp** [comment]
I also like to thank you @gabordemooij – I am using RB since version 3 in almost all PHP projects with database needs.
